### PR TITLE
Fixes #20312: Use bundler_ext in production to load webpack plugin

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@ehelms thanks, does it make sense to add a test to see this file return json correctly?

**Comment by ehelms** [comment]
@ohadlevy I added one, lets see what Jenkins does. Testing the bundler_ext route is not really possible since it requires installing a gem that also changes multiple other paths.

File: test/unit/script/plugin_webpack_directories_test.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'minitest/autorun'
+require 'json'
+
+class AssociationAuthorizerTest < Minitest::Test
+  def setup
+    @root_dir = File.expand_path('../../../..', __FILE__)
```
Comment (ohadlevy) [inline test/unit/script/plugin_webpack_directories_test.rb]: consider using File.join instead (if we ever run on non unix platform).

**Review by dLobatog — CHANGES_REQUESTED** [review]
Setting to WoC based on Hound, `File.join` comment, commit summary length > 65

**Review by dLobatog — APPROVED** [review]

**Review by dLobatog — APPROVED** [review]
Ready to merge if tests pass

File: test/unit/script/plugin_webpack_directories_test.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+require 'json'
+
+class PluginWebpackDirectoriesTest < Minitest::Test
+  def setup
+    @root_dir = File.join('../../../..', __FILE__)
+    @script = File.join("#{@root_dir}/script/plugin_webpack_directories.rb", __FILE__)
+  end
+
+  def test_bundler
+    plugin_webpack = `#{@script}`
```
Comment (dLobatog) [inline test/unit/script/plugin_webpack_directories_test.rb]: Seems that this is not producing any output in Jenkins, as it's nil in the test that is failing

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ehelms Please check out the test, thanks !

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ehelms Please rebase now that the commit by @ohadlevy was merged, thanks!

**Comment by ehelms** [comment]
Rebased and should pass tests this time around

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ehelms Still failing :/

File: test/unit/script/plugin_webpack_directories_test.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require 'minitest/autorun'
+require 'json'
+
+class PluginWebpackDirectoriesTest < Minitest::Test
+  def setup
+    @root_dir = File.join('../../../..', __FILE__)
```
Comment (ohadlevy) [inline test/unit/script/plugin_webpack_directories_test.rb]: nitpicking, but the whole value of using file.join was to not include / or \ so ['..','..','..'....]
also, shouldnt it be 3 times vs 4 ?

if you want to be sure you can probably assert that @root_dir exists is rails root too

**Comment by ehelms** [comment]
@ohadlevy @dLobatog This is now passing tests. Please double check all your requests.

**Comment by ohadlevy** [comment]
overall :+1: but if we add more logic I would prefer to start extracting methods and / or classes etc ...

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ehelms & @ohadlevy !

**Comment by ares** [comment]
I set 1.16 in redmine

**Pull request merged**